IOS Build Error

### Before submitting a new issue

- [x] I tested using the latest version of the library, as the bug might be already fixed.
- [x] I tested using a [supported version](https://github.com/reactwg/react-native-releases/blob/main/docs/support.md) of react native.
- [x] I checked for possible duplicate issues, with possible answers.

### Bug summary

Building gives following error 

error 'withPresentedSheet(completion:)' is only available in iOS 15.0 or newer
error 'detentIdentifierForIndex' is only available in iOS 15.0 or newer
error Failed to build ios project. "xcodebuild" exited with error code '65'. To debug build logs further, consider building your app with Xcode.app

### Library version

2.5.0

### Environment info

    "@lodev09/react-native-true-sheet": "^2.0.5",
    "react-native": "0.75.4",


### Steps to reproduce


using these gives this error

<img width="1120" alt="Image" src="https://github.com/user-attachments/assets/67963ef4-3b56-4833-a78e-6b0f435302c5" />


### Reproducible example repository

none

also 2.0.5 build fails on android for same rn version 
I've tested 2.0.2 build which works for android 


```
Execution failed for task ':lodev09_react-native-true-sheet:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 22s
error Failed to install the app. Command failed with exit code 1: gradlew.bat app:installDevelopmentdebug -PreactNativeDevServerPort=8081
(node:13512) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Warning: Root-level "expo" object found. Ignoring extra keys in Expo config: "name", "displayName"
Learn more: https://expo.fyi/root-expo-object
android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:770:63: warning: Direct call to eval(), but lexical scope is not supported. ...typeof expo === "undefined" ? eval("require") : function (moduleId) { ^~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:1836:18: warning: the variable "DebuggerInternal" was not declared in function "__shouldPauseOnThrow" typeof DebuggerInternal !== 'undefined' && ^~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:14550:7: warning: the variable "setTimeout" was not declared in function "logCapturedError" setTimeout(function () { ^~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:12784:31: warning: the variable "nativeFabricUIManager" was not declared in anonymous function " 296#" var _nativeFabricUIManage = nativeFabricUIManager, ^~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:12812:21: warning: the variable "clearTimeout" was not declared in anonymous function " 296#" cancelTimeout = clearTimeout; ^~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:12825:51: warning: the variable "RN$enableMicrotasksInReact" was not declared in anonymous function " 296#" ... "undefined" !== typeof RN$enableMicrotasksInReact && !!RN$enableMicrotask... ^~~~~~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:12826:47: warning: the variable "queueMicrotask" was not declared in anonymous function " 296#" ...otask = "function" === typeof queueMicrotask ? queueMicrotask : scheduleTi... ^~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:17247:30: warning: the variable "__REACT_DEVTOOLS_GLOBAL_HOOK__" was not declared in anonymous function " 296#" if ("undefined" !== typeof __REACT_DEVTOOLS_GLOBAL_HOOK__) { ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:18966:5: warning: the variable "setImmediate" was not declared in function "handleResolved" setImmediate(function () { ^~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23135:12: warning: the variable "fetch" was not declared in anonymous function " 519#" fetch: fetch, ^~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23136:14: warning: the variable "Headers" was not declared in anonymous function " 519#" Headers: Headers, ^~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23137:14: warning: the variable "Request" was not declared in anonymous function " 519#" Request: Request, ^~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23138:15: warning: the variable "Response" was not declared in anonymous function " 519#" Response: Response ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23295:24: warning: the variable "FileReader" was not declared in function "readBlobAsArrayBuffer" var reader = new FileReader(); ^~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23346:36: warning: the variable "Blob" was not declared in anonymous function " 530#" } else if (support.blob && Blob.prototype.isPrototypeOf(body)) { ^~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23348:40: warning: the variable "FormData" was not declared in anonymous function " 530#" } else if (support.formData && FormData.prototype.isPrototypeOf(body)) { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23350:44: warning: the variable "URLSearchParams" was not declared in anonymous function " 530#" ...e if (support.searchParams && URLSearchParams.prototype.isPrototypeOf(body... ^~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23469:26: warning: the variable "AbortController" was not declared in anonymous function " 536#" var ctrl = new AbortController(); ^~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23603:23: warning: the variable "XMLHttpRequest" was not declared in anonymous function " 540#" var xhr = new XMLHttpRequest(); ^~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:23148:71: warning: the variable "self" was not declared in anonymous function " 522#" ...undefined' && globalThis || typeof self !== 'undefined' && self || ^~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:27180:26: warning: the variable "navigator" was not declared in anonymous function " 698#" "undefined" !== typeof navigator && undefined !== navigator.scheduling && u... ^~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:27290:37: warning: the variable "MessageChannel" was not declared in anonymous function " 698#" };else if ("undefined" !== typeof MessageChannel) { ^~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:27305:34: warning: the variable "nativeRuntimeScheduler" was not declared in anonymous function " 698#" ... = "undefined" !== typeof nativeRuntimeScheduler ? nativeRuntimeScheduler.... ^~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:38827:34: warning: the variable "requestAnimationFrame" was not declared in function "start 9#" ... this._animationFrame = requestAnimationFrame(this.onUpdate.bind(this)); ^~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:60871:12: warning: the variable "_WORKLET" was not declared in function "registerReanimatedError" if (!_WORKLET) { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:62495:94: warning: the variable "location" was not declared in function "registerSensor 1#" ...(0, _PlatformChecker.isWeb)() && location.protocol !== 'https:' ? ' Make s... ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:62819:26: warning: Direct call to eval(), but lexical scope is not supported. workletFun = eval('(' + initData.code + '\n)'); ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:62842:112: warning: the variable "_toString" was not declared in function "valueUnpacker" ...recognized by value unpacker: "${_toString(objectToUnpack)}".`); ^~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:64640:18: warning: the variable "__reanimatedLoggerConfig" was not declared in function "createWorkletRuntime 3#" var config = __reanimatedLoggerConfig; ^~~~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:76990:41: warning: the variable "getComputedStyle" was not declared in function "fixElementPosition" ...entBorderTopValue = parseInt(getComputedStyle(parent).borderTopWidth); ^~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:79519:26: warning: the variable "structuredClone" was not declared in function "createAnimationWithInitialValues" var animationStyle = structuredClone(_config.AnimationsData[animationName... ^~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:79663:5: warning: the variable "document" was not declared in function "configureWebLayoutAnimations" document.getElementById(PREDEFINED_WEB_ANIMATIONS_ID) !== null) { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:79761:27: warning: the variable "HTMLElement" was not declared in function "findDescendantWithExitingAnimation" if (!(node instanceof HTMLElement)) { ^~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:79788:24: warning: the variable "MutationObserver" was not declared in function "addHTMLMutationObserver" var observer = new MutationObserver(function (mutationsList) { ^~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:84989:5: warning: the variable "jest" was not declared in function "beforeTest" jest.useFakeTimers(); ^~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:85822:26: warning: the variable "_getAnimationTimestamp" was not declared in function "computeEasingProgress" var elapsedTime = (_getAnimationTimestamp() - startingTimestamp) / 1000; ^~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:93763:29: warning: the variable "Intl" was not declared in function "formatLanguageCode" formattedCode = Intl.getCanonicalLocales(code)[0]; ^~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:98092:5: warning: the property "isolated" was set multiple times in the object definition. isolated: 'Isolated', ^~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:97758:5: note: The first definition was here. isolated: 'Isolated', ^~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:98141:5: warning: the property "days" was set multiple times in the object definition. days: 'days', ^~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:96834:5: note: The first definition was here. days: 'Days', ^~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:98233:5: warning: the property "type" was set multiple times in the object definition. type: 'Type', ^~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:96739:5: note: The first definition was here. type: 'Type', ^~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:98275:5: warning: the property "day" was set multiple times in the object definition. day: 'Day', ^~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:98140:5: note: The first definition was here. day: 'day', ^~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:98315:5: warning: the property "tp_sl" was set multiple times in the object definition. tp_sl: 'TP/SL', ^~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:97941:5: note: The first definition was here. tp_sl: 'TP/SL', ^~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:98504:5: warning: the property "trailing_stop" was set multiple times in the object definition. trailing_stop: 'Trailing Stop', ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:97942:5: note: The first definition was here. trailing_stop: 'Trailing Stop', ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100172:5: warning: the property "isolated" was set multiple times in the object definition. isolated: 'ایزوله', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:99840:5: note: The first definition was here. isolated: 'ایزوله', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100221:5: warning: the property "days" was set multiple times in the object definition. days: 'روز', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:98916:5: note: The first definition was here. days: 'روز', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100317:5: warning: the property "type" was set multiple times in the object definition. type: 'نوع', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:98821:5: note: The first definition was here. type: 'نوع', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100359:5: warning: the property "day" was set multiple times in the object definition. day: 'روز', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100220:5: note: The first definition was here. day: 'روز', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100399:5: warning: the property "tp_sl" was set multiple times in the object definition. tp_sl: 'سود/ضرر', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100023:5: note: The first definition was here. tp_sl: 'حد سود/حد ضرر', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100401:5: warning: the property "mark" was set multiple times in the object definition. mark: 'مارک', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100109:5: note: The first definition was here. mark: 'مارک', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100588:5: warning: the property "trailing_stop" was set multiple times in the object definition. trailing_stop: 'استاپ متحرک', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100024:5: note: The first definition was here. trailing_stop: 'حد ضرر متحرک', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102209:5: warning: the property "isolated" was set multiple times in the object definition. isolated: 'معزول', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:101877:5: note: The first definition was here. isolated: 'معزول', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102257:5: warning: the property "days" was set multiple times in the object definition. days: 'أيام', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100955:5: note: The first definition was here. days: 'أيام', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102346:5: warning: the property "type" was set multiple times in the object definition. type: 'النوع', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:100860:5: note: The first definition was here. type: 'النوع', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102388:5: warning: the property "day" was set multiple times in the object definition. day: 'اليوم', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102256:5: note: The first definition was here. day: 'يوم', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102428:5: warning: the property "tp_sl" was set multiple times in the object definition. tp_sl: 'جني الأرباح/إيقا ? الخسارة', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102060:5: note: The first definition was here. tp_sl: 'TP/SL', ^~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102430:5: warning: the property "mark" was set multiple times in the object definition. mark: 'السعر الم ?حدد', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102146:5: note: The first definition was here. mark: 'علامة', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102617:5: warning: the property "trailing_stop" was set multiple times in the object definition. trailing_stop: 'إيقا ? الخسارة المتحرك', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102061:5: note: The first definition was here. trailing_stop: 'وق ? الخسارة المتحرك', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104239:5: warning: the property "isolated" was set multiple times in the object definition. isolated: 'İzole', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:103907:5: note: The first definition was here. isolated: 'İzole', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104287:5: warning: the property "days" was set multiple times in the object definition. days: 'gün', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102985:5: note: The first definition was here. days: 'Günler', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104376:5: warning: the property "type" was set multiple times in the object definition. type: 'Tür', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:102890:5: note: The first definition was here. type: 'Tür', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104418:5: warning: the property "day" was set multiple times in the object definition. day: 'Gün', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104286:5: note: The first definition was here. day: 'gün', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104458:5: warning: the property "tp_sl" was set multiple times in the object definition. tp_sl: 'Kâr/Zarar Durdur', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104090:5: note: The first definition was here. tp_sl: 'TP/SL', ^~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104460:5: warning: the property "mark" was set multiple times in the object definition. mark: 'İşaret', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104176:5: note: The first definition was here. mark: 'İşaret', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104647:5: warning: the property "trailing_stop" was set multiple times in the object definition. trailing_stop: 'Takip Eden Zarar Durdur', ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:104091:5: note: The first definition was here. trailing_stop: 'İz Süren Stop', android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:105864:13: warning: the variable "URL" was not declared in function "U 1#" e = URL.createObjectURL(f); ^~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:108466:16: warning: the variable "SharedArrayBuffer" was not declared in function "from" if (typeof SharedArrayBuffer !== 'undefined' && (isInstance(value, Shared... ^~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:118659:18: warning: the variable "localStorage" was not declared in anonymous function " 3419#" return localStorage; ^~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:158099:26: warning: the variable "setInterval" was not declared in function "SessionFlusher" this._intervalId = setInterval(function () { ^~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:158136:9: warning: the variable "clearInterval" was not declared in function "close 5#" clearInterval(this._intervalId); ^~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:175645:48: warning: the variable "CSSRule" was not declared in function "adoptStyleSheets" rules: Array.from(sheet.rules || CSSRule, function (r, index) { ^~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:182311:33: warning: the variable "WeakRef" was not declared in function "addWindow 1#" this.windows.push(new WeakRef(win)); ^~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:182381:11: warning: the variable "cancelAnimationFrame" was not declared in anonymous function " 5657#" cancelAnimationFrame(rafId); ^~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:182467:13: warning: the variable "createImageBitmap" was not declared in anonymous function " 5661#" createImageBitmap(canvas).then(function (bitmap) { ^~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:210846:16: warning: the variable "WebSocket" was not declared in function "getGlobalWebSocket" if (typeof WebSocket !== 'undefined') { ^~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:238322:30: warning: the variable "crypto" was not declared in anonymous function " 7122#" if (typeof crypto != 'undefined' && crypto && (crypto.getRa... ^~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:376457:7: warning: the variable "CSS" was not declared in function "registerCssVariable" CSS.registerProperty({ ^~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:377671:34: warning: the variable "SVGElement" was not declared in function "getElementSize" } else if (target instanceof SVGElement && "getBBox" in target) { ^~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:378053:37: warning: the variable "SVGGraphicsElement" was not declared in function "calcInset" ...else if (current instanceof SVGGraphicsElement && "getBBox" in current) { ^~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:491179:11: warning: the property "style" was set multiple times in the object definition. style: { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:491176:11: note: The first definition was here. style: { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:498192:11: warning: the property "style" was set multiple times in the object definition. style: { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:498188:11: note: The first definition was here. style: { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:501661:17: warning: the property "style" was set multiple times in the object definition. style: { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:501658:17: note: The first definition was here. style: { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:510595:23: warning: the property "style" was set multiple times in the object definition. style: [styles.btnText, { ^~~~~~~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:510591:23: note: The first definition was here. style: { ^~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:520305:13: warning: the property "marginTop" was set multiple times in the object definition. marginTop: 20, ^~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:520301:13: note: The first definition was here. marginTop: 50, ^~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:524146:25: warning: the variable "Atomics" was not declared in anonymous function " 16931#" '%Atomics%': typeof Atomics === 'undefined' ? undefined : Atomics, ^~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:524161:30: warning: the variable "Float16Array" was not declared in anonymous function " 16931#" '%Float16Array%': typeof Float16Array === 'undefined' ? undefined : Float... ^~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:524164:38: warning: the variable "FinalizationRegistry" was not declared in anonymous function " 16931#" ...lizationRegistry%': typeof FinalizationRegistry === 'undefined' ? undefine... ^~~~~~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:528252:13: warning: the property "zIndex" was set multiple times in the object definition. zIndex: 9999999 ^~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:528248:13: note: The first definition was here. zIndex: 11, ^~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:535670:5: warning: the property "itemContainer" was set multiple times in the object definition. itemContainer: { ^~~~~~~~~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:535651:5: note: The first definition was here. itemContainer: { ^~~~~~~~~~~~~~~~ e: file:///C:/myproject/node_modules/@lodev09/react-native-true-sheet/android/src/main/java/com/lodev09/truesheet/core/R~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:535651:5: note: The first definition was here. itemC~~~~~~~~ android\app\build\generated\assets\createBundleDevelopmentDebugJsAndAssets\index.android.bundle:535651:5: note: The first definition was here. itemContainer: { ^~~~~~~~~~~~~~~~ e: file:///C:/myproject/node_modules/@lodev09/react-native-true-sheet/android/src/main/java/com/lodev09/truesheet/core/RootSheetView.kt:63:66 Too many arguments for public open fun handleTouchEvent(p0: MotionEvent, p1: EventDispatcher): Unit defined in com.facebook.react.uimanager.JSTouchDispatcher e: file:///C:/myproject/node_modules/@lodev09/react-native-true-sheet/android/src/main/java/com/lodev09/truesheet/core/RootSheetView.kt:72:66 Too many arguments for public open fun handleTouchEvent(p0: MotionEvent, p1: EventDispatcher): Unit defined in com.facebook.react.uimanager.JSTouchDispatcher FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':lodev09_react-native-true-sheet:compileDebugKotlin'. > A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction > Compilation error. See log for more details * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. > Get more help at https://help.gradle.org. BUILD FAILED in 2m 22s.
```

--- a/ios/TrueSheetView.swift
+++ b/ios/TrueSheetView.swift
@@ -433,9 +433,15 @@ class TrueSheetView: UIView, RCTInvalidating, TrueSheetViewControllerDelegate {
     }
 
     if isPresented {
-      withPresentedSheet { sheet in
-        sheet.selectedDetentIdentifier = viewController.detentIdentifierForIndex(index)
+      if #available(iOS 15.0, *) {
+        withPresentedSheet { sheet in
+          sheet.selectedDetentIdentifier = viewController.detentIdentifierForIndex(index)
 
+          // Trigger onSizeChange event when size is changed while presenting
+          viewControllerDidChangeSize(self.viewController.currentSizeInfo)
+          promise?.resolve(nil)
+        }
+      } else {
         // Trigger onSizeChange event when size is changed while presenting
         viewControllerDidChangeSize(self.viewController.currentSizeInfo)
         promise?.resolve(nil)

